externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala

```diff
@@ -113,4 +113,35 @@ object KyuubiSparkUtil extends Logging {
       SparkSQLEngine.kyuubiConf.get(configEntry)
     }
   }
+
+  private lazy val yarnAppAttemptIdFromEnv: Option[Int] =
+    try {
+      val containerId = System.getenv("CONTAINER_ID")
```

Comment:
Since it is called before SparkContext initialization, we only get it through environments.

refer: https://github.com/apache/spark/blob/9bb358b51e30b5041c0cd20e27cf995aca5ed4c7/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala#L73